src/services/exile_service.py

```diff
@@ -1,34 +1,81 @@
 import discord
 import logging
-from util import log_info_and_embed, add_and_remove_role, send_dm
-from enums import Role
+from util import log_info_and_embed, add_and_remove_role, send_dm, user_has_role
+from enums import Role, ExileStatus
+from database import users_database, exiles_database
+from typing import Optional
+import datetime
+from database.models import Exile, User
 
 logger = logging.getLogger(__name__)
 
 
 async def exile_user(
-    logging_embed: discord.Embed, user: discord.Member, duration: str, reason: str
+    logging_embed: discord.Embed,
+    user: discord.Member,
+    duration: Optional[datetime.timedelta],
+    reason: str,
 ):
-    # TODO implement this functionality
-    log_info_and_embed(
-        logging_embed, logger, f"going to exile {user.mention} for {duration}"
+    # change user role
+    await add_and_remove_role(
+        user, role_to_add=Role.EXILED, role_to_remove=Role.VERIFIED
     )
 
-    # Look up 
-
-    log_info_and_embed(logging_embed, logger, f"because {reason}")
+    # message user
+    await send_dm(
+        user, f"You are being exiled from NAUR FFXIV for the following reason: {reason}"
+    )
 
     # look up user in DB
+    db_user = users_database.get_user(user.id)
+    if db_user is None:
+        log_info_and_embed(
+            logging_embed,
+            logger,
+            f"User not found in database, creating new record",
+        )
+        db_user_id = users_database.add_user(user.id)
+        logger.info(f"Created user record in DB with id {db_user_id}")
 
-    # add exile entry into DB
+        db_user = User(db_user_id, user.id, None, None)
 
-    # change user role
-    await add_and_remove_role(user, role_to_add=Role.EXILED, role_to_remove=Role.VERIFIED)
+    # add exile entry into DB
+    start_timestamp = datetime.datetime.now(datetime.timezone.utc)
+    end_timestamp = None
+    exile_status = ExileStatus.INDEFINITE_EXILE
+    if duration:
+        end_timestamp = start_timestamp + duration
+        exile_status = ExileStatus.TIMED_EXILED
 
+    exile = Exile(None, db_user.user_id, reason, exile_status.value, start_timestamp, end_timestamp)
```

Comment:
is there a reason why the class will contain an exile_id if it's not actually holding that information?

Reply:
I still want to think about if there's a better way to pass around data without bloating the number of classes, but the idea is that we want postgres to take care of generating the actual PK of the row, so it's set up as null when it's first being created

Reply:
that makes sense, i'm just not sure if there's a situation where we will need to throw around the exile id around except for getting the list of exiles. it's not detailed in the spec right now so i'm not sure if we're actually going to be doing that